airflow/operators/latest_only_operator.py

```diff
@@ -46,7 +46,7 @@ def choose_branch(self, context: Dict) -> Union[str, Iterable[str]]:
         if context['dag_run'] and context['dag_run'].external_trigger:
             self.log.info(
                 "Externally triggered DAG_Run: allowing execution to proceed.")
-            return context['task'].get_direct_relative_ids(upstream=False)
+            return list(context['task'].get_direct_relative_ids(upstream=False))
```

Comment:
Why do we need a `list` instead of a `set` ??

Reply:
`TypeError: Object of type 'set' is not JSON serializable`

Reply:
To be honest, I was quite surprised, I think this is also something that we should backport to 1.11 otherwise this operator works only with pickle xcom. @kaxil

Reply:
Do we get this error in Xcom.serialize, maybe we should have something in Xcom.serialize to handle serialization of set as there would be possibly be many other places where set is used. WDYT ?

Reply:
I was thinking about that and I am against it. JSON has no `set` structure and users should be aware of this. Additionally we want to support `set` in general then we have to support all possible nested structures (dicts + sets, list of sets, etc)

Reply:
We do that for SerializedDag, we might be able to use the same code for it

Reply:
That is out of scope of this PR but can you please create a Github issues for this please?

i.e. Support for Set and other nested structures for Xcom serialization

Reply:
Sure https://github.com/apache/airflow/issues/8703